### PR TITLE
mimirtool: fail on duplicate rules with --strict

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

Changes the behaviour of `mimirtool rules check` to fail on duplicated rules when `--strict` is provided.

Also changed the output to use the logger instead of fmt, and added a test.

The output looks like this (on main before https://github.com/grafana/mimir/pull/5023 was merged):

```
$ go run ./cmd/mimirtool rules check --strict --rule-dirs ./operations/mimir-mixin-compiled
ERRO[0000] duplicate rules found                         duplicate_rules=2 error="rules should emit unique series, to avoid producing inconsistencies while recording expressions" namespace=alerts
ERRO[0000] duplicate rule                                label[severity]=warning metric=MimirRolloutStuck namespace=alerts
ERRO[0000] duplicate rule                                label[severity]=critical metric=MimirCompactorHasNotUploadedBlocks namespace=alerts
mimirtool: error: 2 duplicate rules found in namespace "alerts", try --help
exit status 1
```

#### Which issue(s) this PR fixes or relates to

Relates to https://github.com/grafana/mimir/pull/5023

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/mimirtool/commands/rules_test.go [hunk 1]
```diff
@@ -115,6 +117,105 @@ func TestCheckDuplicates(t *testing.T) {
...
+			shouldFail: false,
+		},
+		{
+			name:       "bad rule name, strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
+			strict:     true,
+			shouldFail: true,
+		},
+		{
+			name:       "strictly rule name, not strict",
```
Comment (aknuds1) [inline pkg/mimirtool/commands/rules_test.go:162]:
```suggestion
			name:       "strictly bad rule name, not strict",
```

File: pkg/mimirtool/commands/rules_test.go [hunk 2]
```diff
@@ -115,6 +117,105 @@ func TestCheckDuplicates(t *testing.T) {
...
+					Namespace: "test",
+					Groups: []rwrulefmt.RuleGroup{{
+						RuleGroup: rulefmt.RuleGroup{
+							Name:  "rulegroup",
+							Rules: tc.rules,
+						},
+					}},
+				}
+				require.NoError(t, yaml.NewEncoder(f).Encode(contents))
+				require.NoError(t, f.Close())
```
Comment (aknuds1) [inline pkg/mimirtool/commands/rules_test.go:202]: I suggest you use `t.Cleanup` instead. I don't think the test should fail either if closing the file fails for whatever reason.
```suggestion
```

**Review by aknuds1 — COMMENTED** [review]
The implementation looks solid, but there are some minor issues with the tests.

File: pkg/mimirtool/commands/rules_test.go [hunk 3]
```diff
@@ -115,6 +117,105 @@ func TestCheckDuplicates(t *testing.T) {
...
+			shouldFail: true,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			file := filepath.Join(t.TempDir(), "rules.yaml")
+			{
+				// Setup.
+				f, err := os.Create(file)
+				require.NoError(t, err)
+				contents := rules.RuleNamespace{
```
[the remark this answers is not in the document]
Comment (colega) [reply pkg/mimirtool/commands/rules_test.go:194]: Right, added it through an own commit (`f.Close()` returns an error)

File: pkg/mimirtool/commands/rules_test.go (hunk 2 above)
Comment (colega) [reply pkg/mimirtool/commands/rules_test.go:204]: I want to make sure that file is closed before we read it, so leaving this here.

File: pkg/mimirtool/commands/rules_test.go [hunk 4]
```diff
@@ -115,6 +117,105 @@ func TestCheckDuplicates(t *testing.T) {
...
+		},
+		{
+			name:       "bad rule name, strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
+			strict:     true,
+			shouldFail: true,
+		},
+		{
+			name:       "strictly rule name, not strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
```
[the remark this answers is not in the document]
Comment (colega) [reply pkg/mimirtool/commands/rules_test.go:157]: Right, removed that one.

**Comment by colega** [comment]
Thank you for your review, @aknuds1, the comments should be addressed now.

File: pkg/mimirtool/commands/rules_test.go (hunk 2 above)
Comment (aknuds1) [reply pkg/mimirtool/commands/rules_test.go:198]: Makes sense.

**Review by aknuds1 — APPROVED** [review]
LGTM, thanks for addressing my feedback!

**Pull request merged**